src/renderer/tvgText.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2023 the ThorVG project. All rights reserved.
+
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+
+#include "tvgText.h"
+
+
+/************************************************************************/
+/* Internal Class Implementation                                        */
+/************************************************************************/
+
+
+
+/************************************************************************/
+/* External Class Implementation                                        */
+/************************************************************************/
+
+
+Text::Text() : pImpl(new Impl)
+{
+    Paint::pImpl->id = TVG_CLASS_ID_TEXT;
+}
+
+
+Text::~Text()
+{
+    delete(pImpl);
+}
+
+
+Result Text::text(const char* text) noexcept
+{
```

Comment:
There is no reason to do it. Let's allow a null for removing the existing text.

Reply:
OK... I think we should also check for text that doesn't end in \0.
```cpp
    char* no_end_of_string = (char*) malloc(sizeof(char) * 3);
    no_end_of_string[0] = 'A';
    no_end_of_string[1] = 'B';
    no_end_of_string[2] = 'C';

    auto text = tvg::Text::gen();
    text->font("Arial", 80);
    text->text(no_end_of_string);
    text->fill(255, 255, 255);
    canvas->push(std::move(text));
```

Reply:
We don't need to address potential misuse in C. In such cases, functions like `strdup()`, `strlen()`, and other string-related functions would fail. This isn't an issue with ThorVG itself. Even if we attempt to prevent it, crashes could still occur elsewhere  outside  due to the wrong usage. They can use the proper way.

```
text = strdup("ABC");
char text[] ="ABC";
string text ="ABC";
```

Reply:
I thought we would agree that string data can cause a variety of potential problems.
and I think handling exceptions is different from a conceptual problem.
The text() method returns tvg::Result. it represents some problems we anticipate in advance.
```
InvalidArguments,      ///< The value returned in the event of a problem with the arguments given to the API - e.g. empty paths or null pointers
```
If we always expect the string input to the text() method to be valid, we don't even need to return Success at least in the current situation.
anyway,ok i understand what you want